### ISSUE TITLE
Type error: 'searchParams' is possibly 'null'

### ISSUE THREAD

**Issue by olivierlambert** [calcom/cal.com#14844]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Doing a yarn build on latest release is triggering this:
```
@calcom/web:build: Failed to compile.
@calcom/web:build:
@calcom/web:build: ../../packages/features/ee/workflows/components/WorkflowDetailsPage.tsx:48:23
@calcom/web:build: Type error: 'searchParams' is possibly 'null'
@calcom/web:build
@calcom/web:build:   46 |
@calcom/web:build:   47 |   const searchParams = useSearchParams();    
@calcom/web:build: > 48 |   const eventTypeId = searchParams.get("eventTypeId"); 
@calcom/web:build:      |                       ^    
@calcom/web:build:   49 |  
@calcom/web:build:   50 |   const eventTypeOptions = useMemo(    
@calcom/web:build:   51 |     () =>         
@calcom/web:build: ERROR: command finished with error: command (/data/cal/apps/web) yarn run build exited (1)                                                                                                    command (/data/cal/apps/web) yarn run build exited (1)  
Tasks:    6 successful, 7 total 
Cached:    2 cached, 7 total  
Time:    2m8.055s
Failed:    @calcom/web#build
ERROR  run failed: command  exited (1)
```

### Steps to Reproduce

1. git checkout latest release
2. yarn then yarn build
3. get the issue

### Actual Results

A broken build

### Expected Results

A working build

### Technical details

- Browser version: not relevant here
- Node.js version: v18.19.0

### Evidence

- Build worked on the release 4.0.1

**Comment by olivierlambert** [comment]
Applying the change suggested by the bot makes the build work. However I think this should be fixed upstream ASAP, since I'm probably not the only one with the issue? Thanks! 👍

**Comment by Conni2461** [comment]
i think this is fixed in ef981055030f7ec7c5b4550b27748bfc968510a4

i currently also ran into this issue while trying to deploy a v4.0.5 build to production.

**Comment by olivierlambert** [comment]
Fixed by latest release `v4.0.6`. Closing.
